CountingSort/app_resources/scatter_shader.comp.hlsl

```diff
@@ -0,0 +1,80 @@
+#include "nbl/builtin/hlsl/bda/__ptr.hlsl"
+#include "nbl/builtin/hlsl/sort/counting.hlsl"
+#include "nbl/builtin/hlsl/sort/common.hlsl"
+
+[[vk::push_constant]] nbl::hlsl::sort::CountingPushData pushData;
+
+struct PtrAccessor
+{
+    static PtrAccessor createAccessor(uint64_t addr)
+    {
+        PtrAccessor ptr;
+        ptr.addr = addr;
+        return ptr;
+    }
+
+    uint32_t get(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().load();
+    }
+
+    void set(uint64_t index, uint32_t value)
+    {
+        bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().store(value);
+    }
+
+    bda::__spv_ptr_t<uint32_t> get_ptr(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().get_ptr();
+    }
+
+    uint64_t addr;
+};
+
+struct DoublePtrAccessor
+{
+    static DoublePtrAccessor createAccessor(uint64_t in_addr, uint64_t out_addr)
+    {
+        DoublePtrAccessor ptr;
+        ptr.in_addr = in_addr;
+        ptr.out_addr = out_addr;
+        return ptr;
+    }
+
+    uint32_t get(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (in_addr + sizeof(uint32_t) * index).template
+        deref().load();
+    }
+
+    void set(uint64_t index, uint32_t value)
+    {
+        bda::__ptr < uint32_t > (out_addr + sizeof(uint32_t) * index).template
+        deref().store(value);
+    }
+
+    uint64_t in_addr, out_addr;
+};
```

Comment:
if you made these templated I could let you slap them in a header like 
https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/hlsl/memory_accessor.hlsl

and call them `bda_accessor`

but they';d need to:
1. be templated
2. guard the `atomicAdd` method with an `enable_if` checking for integral type and being 4 or 8 bytes in size

Reply:
Done